inst/report/scripts/generate_report.R

```diff
@@ -0,0 +1,705 @@
+# Author(s): Awa Synthia
+# Last modified: 2024
+
+# get fasta of pathogen and/or drug
+#' @export
+get_card_data <- function(pathogen = NULL, drug = NULL) {
+  destination_dir <- "CARD_data"
+  # Check if CARD data exists
+  if (!dir.exists(destination_dir)) {
+    dir.create(destination_dir)
+  }
+  if (!file.exists("CARD_data/aro_index.tsv")) {
+    # Step 1: Download CARD Data
+    download.file("https://card.mcmaster.ca/latest/data", "card_data.tar.bz2")
+    #unzip("card_data.tar.bz2", exdir = "CARD_data")
+    system(paste("tar -xjf card_data.tar.bz2 -C", destination_dir))
+  }
+
+  # Step 2: Open ARO_index.tsv
+  aro_data <- read.delim("CARD_data/aro_index.tsv", sep = "\t", header = TRUE)
+
+  # Step 3: Map CARD Short Name
+  antibiotics <- read.delim("CARD_data/shortname_antibiotics.tsv",
+                            sep = "\t", header = TRUE)
+  pathogens <- read.delim("CARD_data/shortname_pathogens.tsv",
+                          sep = "\t", header = TRUE)
+
+  aro_data <- aro_data %>%
+    mutate(
+      pathogen = str_extract(CARD.Short.Name, "^[^_]+"),
+      gene = str_extract(CARD.Short.Name, "(?<=_)[^_]+"),
+      drug = str_extract(CARD.Short.Name, "(?<=_)[^_]+$")
+    ) %>%
+    left_join(pathogens, by = c("pathogen" = "Abbreviation")) %>%
```

Comment:
Could you check if data.table::merge or `dtplyr` is faster (and more efficient) than dplyr's join functions? Join is definitely better than baseR's merge, but just checking if there's a better solution when we have 10s of 1000s of rows!